internal/hooks/evaluation_execution.go

```diff
@@ -0,0 +1,68 @@
+package hooks
+
+import (
+	gocontext "context"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldlog"
+	"github.com/launchdarkly/go-sdk-common/v3/ldreason"
+	"github.com/launchdarkly/go-server-sdk/v7/ldhooks"
+)
+
+// EvaluationExecution represents the state of a running series of evaluation stages.
+type EvaluationExecution struct {
```

Comment:
I changed the model so the runner runs an execution. This should be slightly less copying of the parameters as well.